astropy/units/quantity.py

```diff
@@ -156,6 +156,48 @@ def _construct_from_dict(self, map):
         value = map.pop('value')
         return self._parent_cls(value, **map)
 
+    def empty_like(self, cols, length, metadata_conflicts='warn', name=None):
+        """
+        Return an empty Quantity instance which is consistent with the
+        input ``cols`` and has ``length`` rows.  This is intended for creating
+        an empty column object whose elements can be set in-place for
+        table operations like join or vstack.
+
+        Parameters
+        ----------
+        cols : list
+            List of input columns
+        length : int
+            Length of the output column object
+        metadata_conflicts : str ('warn'|'error'|'silent')
+            How to handle metadata conflicts.
+        name : str
+            Output column name
+
+        Returns
+        -------
+        col : Quantity (or subclass)
+            Empty version of this class consistent with ``cols``
+        """
+
+        # Get merged info attributes like shape, dtype, format, description, etc.
+        attrs = self.merge_cols_attributes(cols, metadata_conflicts, name,
+                                           ('meta', 'format', 'description'))
+
+        # Make an empty quantity filled with Nan using the unit of the last one.
+        shape = (length,) + attrs.pop('shape')
+        dtype = attrs.pop('dtype')
+        data = np.empty(shape=shape, dtype=dtype)
```

Comment:
It is needed to fill it with `nan`? Or are you setting up for having that mean masked? In any case, a short-cut is 
```
data = np.full(shape, np.nan, dtype=dtype)
```